uspace/app/taskbar/wndlist.c

```diff
@@ -716,13 +716,27 @@ errno_t wndlist_repaint(wndlist_t *wndlist)
 static void wndlist_button_clicked(ui_pbutton_t *pbutton, void *arg)
 {
 	wndlist_entry_t *entry = (wndlist_entry_t *)arg;
+	wndmgt_window_info_t *winfo = NULL;
 	sysarg_t dev_id;
+	errno_t rc;
+	bool minimized = false;
 
 	/* ID of device that clicked the button */
 	dev_id = entry->wndlist->ev_idev_id;
 
-	(void) wndmgt_activate_window(entry->wndlist->wndmgt,
-	    dev_id, entry->wnd_id);
+	rc = wndmgt_get_window_info(entry->wndlist->wndmgt,
+	    entry->wnd_id, &winfo);
+	if (rc == EOK) {
+		minimized = (winfo->flags & wndf_minimized) == 0;
```

Comment:
Note that the logic is reversed w.r.t. the name of the variable: minimized will be true iff the window is not minimized